src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -0,0 +1,640 @@
+    }
+
+    private void HistoryListMenuIndexChanged(int index)
+    {
+        if (generationDisplayed == index && speciesListMenu.Popup.GetItemCount() > 0)
+            return;
+
+        historyListMenu.Text = index.ToString();
+
+        generationDisplayed = index;
+        UpdateAutoEvoReport();
+        UpdateSpeciesList();
+    }
+
+    private void UpdateAutoEvoReport()
+    {
+        if (generationDisplayed < runResultsList.Count && generationDisplayed > -1)
+        {
+            autoEvoResultsLabel.ExtendedBbcode = runResultsList[generationDisplayed].ToString();
+        }
+    }
+
+    private void UpdateSpeciesList()
+    {
+        speciesListMenu.ClearAllItems();
+
+        foreach (var pair in speciesHistoryList[generationDisplayed].OrderBy(p => p.Value.FormattedName))
+        {
+            speciesListMenu.AddItem(pair.Value.FormattedName, false, Colors.White);
+        }
+
+        speciesListMenu.CreateElements();
+    }
+
+    private void SpeciesListMenuIndexChanged(int index)
+    {
+        var speciesName = speciesListMenu.Popup.GetItemText(index);
+        var species = speciesHistoryList[generationDisplayed].Values.First(p => p.FormattedName == speciesName);
+
+        if (species == speciesPreview.PreviewSpecies)
+            return;
+
+        UpdateSpeciesPreview(species);
+    }
+
+    private void UpdateSpeciesPreview(Species species)
+    {
+        speciesListMenu.Text = species.FormattedName;
+        speciesPreview.PreviewSpecies = species;
+        hexPreview.PreviewSpecies = species as MicrobeSpecies;
+        UpdateSpeciesDetail(species);
+    }
+
+    private void EvolutionaryTreeNodeSelected(int generation, uint id)
+    {
+        HistoryListMenuIndexChanged(generation);
+        UpdateSpeciesPreview(speciesHistoryList[generation][id]);
+    }
+
+    private void UpdateSpeciesDetail(Species species)
+    {
+        /*
+        speciesDetailsLabel.ExtendedBbcode = $"[b]Species:[/b]\n  {species.FormattedNameBbCode}:{species.ID}\n" +
+            $"[b]Generation:[/b]\n  {species.Generation}\n" +
+            $"[b]Population:[/b]\n  {species.Population}\n" +
+            $"[b]Colour:[/b]\n  #{species.Colour.ToHtml()}\n" +
+            $"[b]Behaviour[/b]\n  {species.Behaviour.Join(b => b.Key + ": " + b.Value, "\n  ")}\n";
+
+        switch (species)
+        {
+            case MicrobeSpecies microbeSpecies:
+            {
+                speciesDetailsLabel.ExtendedBbcode += $"[b]Stage:[/b]\n  Microbe\n" +
+                    $"[b]Membrane Type:[/b]\n  {microbeSpecies.MembraneType.Name}\n" +
+                    $"[b]Membrane Rigidity:[/b]\n  {microbeSpecies.MembraneRigidity}\n" +
+                    $"[b]Base Speed:[/b]\n  {microbeSpecies.BaseSpeed}\n" +
+                    $"[b]Base Rotation Speed[/b]\n  {microbeSpecies.BaseRotationSpeed}\n" +
+                    $"[b]Base Hex Size[/b]\n  {microbeSpecies.BaseHexSize}";
+                break;
+            }
+        }
+        */
```

Comment:
This should be removed.

Reply:
I think this makes the strange formatSafe clearer.

Reply:
I have not been keeping up anywhere with showing the actual text that a translation uses. Only a few places one older developer who I think is now inactive used. I think that's a huge antipattern because the text is changed on weblate so no one is going to update the code comments which will become outdated, and outdated comments are worse than no comments.